lib/modules/manager/bazel-module/extract.ts

```diff
@@ -0,0 +1,22 @@
+import { logger } from '../../../logger';
+import { isNotNullOrUndefined } from '../../../util/array';
+import type { PackageFileContent } from '../types';
+import { toPackageDependency } from './bazel-dep';
+import { parse } from './parser';
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string
+): PackageFileContent | null {
+  const deps = parse(content)
+    .map((frag) => toPackageDependency(frag))
+    .filter(isNotNullOrUndefined);
+  if (deps.length) {
+    return { deps };
+  }
+  logger.debug(
+    { packageFile },
+    'The package file did not contain any package dependencies.'
+  );
+  return null;
```

Comment:
Just to illustrate my point:

```suggestion
  const raw = parse(content);
  return LooseArray(BazelDepChildrenSchema)
    .transform((deps) => (deps.length ? { deps } : null))
    .parse(raw);
```

We have two phases:
1. From string to unstructured JSON data (`good-enough-parser`)
2. From unstructured data to the target interfaces (`zod`)

Transforms and pipes at the phase 2 should prevent us from defensive type guarding, so I hope there won't be a need for `as` methods.

Reply:
(`LooseArray` is the array with values that don't match the schema filtered out, with optional error reporting)